File: Algorithms/String/6. ZigZag Conversion.py
class Solution(object):
	def convert(self, s, numRows):
		if(numRows <= 1):
			return s
		string = [''] * numRows
		row = 0
		for inx, char in enumerate(s):
			if(row == 0):
				step = 1
			elif(row == numRows - 1):
				step = -1
			string[row] += char
			row += step
		ans = ''
		for i in string:
			ans += i
		return ans

File: Algorithms/String/6. ZigZag Conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
		if(numRows <= 1) {
			return s;
		}
		string str[numRows];
		int step = 0, row = 0;
		for(int inx = 0; inx < s.size(); inx += 1) {
			str[row] += s[inx];
			if(row == 0) {
				step = 1;
			}
			else if(row == numRows-1) {
				step = -1;
			}
			row += step;
		}
		string result;
		for(auto i : str) {
			result += i;
		}
		return result;
    }
};
